sdk/resource/resource_test.go

```diff
@@ -109,18 +120,51 @@ func TestMerge(t *testing.T) {
 		t.Run(fmt.Sprintf("case-%s", c.name), func(t *testing.T) {
 			res := resource.Merge(c.a, c.b)
 			if diff := cmp.Diff(
-				sortedAttributes(res.Attributes()),
-				sortedAttributes(c.want),
+				res.Attributes(),
+				c.want,
 				cmp.AllowUnexported(core.Value{})); diff != "" {
 				t.Fatalf("unwanted result: diff %+v,", diff)
 			}
 		})
 	}
 }
 
-func sortedAttributes(attrs []core.KeyValue) []core.KeyValue {
-	sort.Slice(attrs[:], func(i, j int) bool {
-		return attrs[i].Key < attrs[j].Key
-	})
-	return attrs
+func TestString(t *testing.T) {
+	for _, test := range []struct {
+		kvs  []core.KeyValue
+		want string
+	}{
+		{
```

Comment:
Test for a Resource that has no keys
```suggestion
		{
			kvs:  nil,
			want: "Resource()",
		},
		{
```